jetbrains-core/src/software/aws/toolkits/jetbrains/settings/AwsApplicationSettings.kt

```diff
@@ -0,0 +1,35 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.settings
+
+import com.intellij.openapi.components.PersistentStateComponent
+import com.intellij.openapi.components.ServiceManager
+
+class AwsApplicationSettings : PersistentStateComponent<AwsApplicationConfiguration> {
+    private var state = AwsApplicationConfiguration()
+
+    override fun getState(): AwsApplicationConfiguration {
+        return state
+    }
+
+    override fun loadState(state: AwsApplicationConfiguration) {
+        this.state = state
+    }
+
+    var samLocation: String
+        get() = state.samLocation
+        set(value) {
+            state.samLocation = value
+        }
+
+    companion object {
+        @JvmStatic
+        fun getInstance(): AwsApplicationSettings = ServiceManager.getService(
+            AwsApplicationSettings::class.java)
+    }
+}
+
+data class AwsApplicationConfiguration(
+    var samLocation: String = ""
```

Comment:
nit: can we can something more descriptive than `location` it's the executable path right?

Reply:
samExecutablePath?